### ISSUE TITLE
Strange cluster problem

### ISSUE THREAD

**Issue by jones-michael-s** [rhodricusack/automaticanalysis#126]
Hello all,

I have a strange cluster problem I’m trying to troubleshoot. 

Usually when I submit an aa job on our cluster I get a log file that looks like this:

```
--------------------------------------------------------------
Begin PBS Prologue Wed Aug 30 11:09:18 CDT 2017 1504109358
Job ID:			3268574.mgt2.cluster
Username:		jonesm
Group:			jpeelle
Nodes:			node10
End PBS Prologue Wed Aug 30 11:09:18 CDT 2017 1504109358
--------------------------------------------------------------
Executing: /act/matlab/R2016b/bin/worker 

                            < M A T L A B (R) >
                  Copyright 1984-2016 The MathWorks, Inc.
                   R2016b (9.1.0.441655) 64-bit (glnxa64)
                             September 7, 2016

Warning: Function mkdir has the same name as a MATLAB builtin. We suggest you
rename the function to avoid a potential name conflict. 
 
To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.
 
2017-08-30 11:13:56 | About to exit MATLAB normally
2017-08-30 11:13:56 | About to exit with code: 0
--------------------------------------------------------------
Begin PBS Epilogue Wed Aug 30 11:14:01 CDT 2017 1504109641
Job ID:			3268574.mgt2.cluster
Username:		jonesm
Group:			jpeelle
Job Name:		Job26.1
Limits:       	mem=1000mb,neednodes=1:ppn=1,nodes=1:ppn=1,walltime=01:00:00
Resources:	cput=00:04:19,energy_used=0,mem=765032kb,vmem=3904224kb,walltime=00:04:38
Nodes:	 		node10
End PBS Epilogue Wed Aug 30 11:14:02 CDT 2017 1504109642
--------------------------------------------------------------

```
(The mkdir warning is because I had to roll my own mkdir function to get around a Matlab mkdir bug).

My problem is I’m frequently getting jobs that abort immediately. The log file looks like this:

```
--------------------------------------------------------------
Begin PBS Prologue Wed Aug 30 11:08:41 CDT 2017 1504109321
Job ID:			3268572.mgt2.cluster
Username:		jonesm
Group:			jpeelle
Nodes:			node10
End PBS Prologue Wed Aug 30 11:08:41 CDT 2017 1504109321
--------------------------------------------------------------
Executing: /act/matlab/R2016b/bin/worker 
--------------------------------------------------------------
Begin PBS Epilogue Wed Aug 30 11:08:47 CDT 2017 1504109327
Job ID:			3268572.mgt2.cluster
Username:		jonesm
Group:			jpeelle
Job Name:		Job24.1
Limits:     	mem=1000mb,neednodes=1:ppn=1,nodes=1:ppn=1,walltime=01:00:00
Resources:	cput=00:00:00,energy_used=0,mem=0kb,vmem=0kb,walltime=00:00:01
Nodes:	       	node10
End PBS Epilogue Wed Aug 30 11:08:47 CDT 2017 1504109327
--------------------------------------------------------------

```

It looks like the node simply rejects the job (note the walltime). There’s not even a Matlab splash message generated.

The strange thing is it will fail on a random module in the tasklist. If I restart, aa will pick up where it left off and continue the job normally. Sometimes it finishes, or sometime it fails again at a later stage. Weirder still, if I clear out all the results and restart from scratch, sometimes the entire analysis finishes without any problems at all.

I’ve tried adjusting the requested walltime and memory but no joy. I’ve submitted at different times. Cluster load doesn’t seem to have an effect. I’ve even tried putting a bunch of pause( ) statements in aaq_qsub (maybe submitting too many jobs too quickly?). Nothing helps.

Anyone have any idea what could cause this, or have any suggestions for troubleshooting?  There’s no diagnostic or error messages in any of the job files. 

Thanks much,
Mike

**Comment by tiborauer** [comment]
It is hard to tell when it is not replicable. We (at the CBSU) also had similar issues. Darren (@dprice80)'s recent PR is hoped to solve it (among other things) by using a more efficient internal job scheduling.

**Comment by jones-michael-s** [comment]
Well, I guess it's useful to know others have seen the problem.

I'll try the new scheduler once Darren's PR is merged.

**Comment by dprice80** [comment]
This could be caused by any number of things but my new code should take care of it since its designed to take care of intermittent server faults. Do let me know if it does not solve your problem.

**Comment by jones-michael-s** [comment]
Thanks Darren. 

I pulled aaq_qsub.m from your forked repository and just tried it but the code crashed. This probably means there's a few other new files I need to pull as well. Can you point me to the minimal set that involves the new scheduler? (Unfortunately, I can't just pull the whole thing -- we're running under OS X which involves some custom fixes to aa that I haven't yet put into the main repo...)

**Comment by dprice80** [comment]
Sorry I'm not sure which files you would need to get things working. My
repo is up to date with main, so you should just merge my version into
yours. You would need to do that before submitting your pull request
anyway. The trouble is I rebased by branch (I didn't know what I was doing
at the time), so all of main branch commits were mixed in with my history
(I won't be doing that again). So it's not possible to go back to a time
when none of tibor/johan's changes were not applied to my branch.

Darren

**Comment by dprice80** [comment]
Hi Mike. My latest changes have now been merged into the main repo. Give it a try and let me know whether this solved your problem. Would be good to know if there are any cases that are not covered by my error handling.

**Comment by jones-michael-s** [comment]
Thanks Darren.

We have a few deadlines on other projects pressing at the moment, but I'll get back to aa testing soon.

**Comment by dprice80** [comment]
Hi Mike. So was this issue solved with the update?

**Comment by jones-michael-s** [comment]
I have not seen a problem after switching to the new scheduler code. We are about to unleash hell on our cluster with a new project, so that will give the new code a more vigorous shake-down. I remain optimistic.

**Comment by dprice80** [comment]
I look forward to hearing about what is unleashed from those fiery pits 😀
